services/filesstore/s3store_test.go

```diff
@@ -7,26 +7,24 @@ import (
 	"testing"
 
 	"github.com/mattermost/mattermost-server/model"
+	"github.com/stretchr/testify/require"
 )
 
 func TestCheckMandatoryS3Fields(t *testing.T) {
 	cfg := model.FileSettings{}
 
 	err := CheckMandatoryS3Fields(&cfg)
-	if err == nil || err.Message != "api.admin.test_s3.missing_s3_bucket" {
-		t.Fatal("should've failed with missing s3 bucket")
-	}
+	res := err == nil || err.Message != "api.admin.test_s3.missing_s3_bucket"
+	require.False(t, res, "should've failed with missing s3 bucket")
 
 	cfg.AmazonS3Bucket = model.NewString("test-mm")
 	err = CheckMandatoryS3Fields(&cfg)
-	if err != nil {
-		t.Fatal("should've not failed")
-	}
+	require.Nil(t, err, "should've not failed")
```

Comment:
This error string doesn't give any extra information, you can remove it.

Reply:
Fail message removed as you suggested.